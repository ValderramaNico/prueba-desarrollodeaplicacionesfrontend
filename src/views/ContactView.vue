<template>
  <div>
    <h1>contact</h1>
    <nav>
      <router-link to="/">ir al Home</router-link>
    </nav>
  </div>
</template>
