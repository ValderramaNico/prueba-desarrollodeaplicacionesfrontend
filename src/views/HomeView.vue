<template>
  <div class="home">
    <h1>Pruebas</h1>
    <hr />
    <br />
    <section>
      <h2 class="titles">Ejercicio 1</h2>
      <ContadorComponent />
    </section>
    <br />
    <hr />
    <section>
      <h2 class="titles">Ejercicio 2</h2>
      <h3 for="">Rutas del "Router"</h3>
      <nav>
        <router-link to="/">Home</router-link> |
        <router-link to="/about">About</router-link> |
        <router-link to="/contact">Contact</router-link>
      </nav>
    </section>
    <br />
    <hr />
    <section>
      <h2 class="titles">Ejercicio 3</h2>
      <ParentComponent />
    </section>
    <br />
    <hr />
    <FooterComponent />
  </div>
</template>

<script>
import ContadorComponent from "@/components/ContadorComponent.vue";
import ParentComponent from "@/components/ParentComponent.vue";
import FooterComponent from "@/components/FooterComponent.vue";

export default {
  name: "HomeView",
  components: {
    ContadorComponent,
    ParentComponent,
    FooterComponent
  },
};
</script>

<style>
.titles {
  margin-left: 4rem;
  text-align: left;
}
</style>
