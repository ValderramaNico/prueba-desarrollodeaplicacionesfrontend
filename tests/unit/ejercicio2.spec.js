import { mount } from "@vue/test-utils";
import HomeView from "@/views/HomeView.vue";
import AboutView from "@/views/AboutView.vue";
import ContactView from "@/views/ContactView.vue";
import { createRouter, createWebHistory } from "vue-router";

//* ------------------------------------------------------------------------------------------ *//

//* Test para la ruta y el componente About, se monta la vista 'AboutView' y se hace un 'SnapShot' *//
//* que queda registrado en el archivo 'ejercicio2.spec.js.snap' *//

//Esta prueba compara el HTML actual de la vista 'AboutView' con un Snapshot previo.
describe("Test vista 'About'", () => {
  it("Test 1 - Snapshot vista 'About' succesful", () => {
    const wrapper = mount(AboutView);
    expect(wrapper.html()).toMatchSnapshot();
  });
  //Esta prueba verifica la existencia del componente 'AboutView' en la ruta '/about'.
  it("Test 1.2 - Prueba existencia del componente en la ruta", async () => {
    const routes = [
      //Configuración de la ruta que se va a probar
      {
        path: "/about", //Se entrega la URL
        name: "about", //Se entrega el nombre de la ruta
        component: () => AboutView, //Se entrega el componente Vue que se va a renderizar
      },
    ];

    const router = createRouter({
      //Creación del router con la configuración de rutas
      history: createWebHistory(process.env.BASE_URL), //Configura el historial para la navegación
      routes, //Registra las rutas definidas
    });

    router.push("/about"); //Navega a la ruta '/about' en el router
    await router.isReady(); //Espera a que la navegación a la ruta esté completa

    const wrapper = mount(AboutView, {
      //Monta el componente 'AboutView' para la prueba
      global: {
        //Configura elementos globales en el entorno de prueba
        plugins: [router], //Registra el plugin `router` para que esté disponible en el componente
      },
    });

    expect(wrapper.findComponent(AboutView).exists()).toBe(true); //Verifica que el componente 'AboutView' exista en la ruta
    expect(wrapper.html()).toMatchSnapshot(); //Compara el HTML actual de la vista 'AboutView' con un Snapshot previo
  });
});

//* ------------------------------------------------------------------------------------------ *//

//* Test para la ruta y el componente Contact, se monta la vista 'ContactView' y se hace un 'SnapShot' *//
//* que queda registrado en el archivo 'ejercicio2.spec.js.snap' *//
//El proceso es el mismo que en el test para la ruta y componente 'About'

describe("Test 2 de vista 'Contact'", () => {
  it("Test 2.1 - Snapshot vista 'Contact'", () => {
    const wrapper = mount(ContactView);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Test2 - Prueba existencia del componente en la ruta", async () => {
    const routes = [
      {
        path: "/contact",
        name: "contact",
        component: () => ContactView,
      },
    ];
    const router = createRouter({
      history: createWebHistory(process.env.BASE_URL),
      routes,
    });

    router.push("/contact");
    await router.isReady();

    const wrapper = mount(ContactView, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.findComponent(ContactView).exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
});

//* ------------------------------------------------------------------------------------------ *//

//* Test para la ruta y el componente Home, se monta la vista 'HomeView' y se hace un 'SnapShot' *//
//* que queda registrado en el archivo 'ejercicio2.spec.js.snap' *//
//El proceso es el mismo que en el test para la ruta y componente 'About'

describe("Test vista 'Home'", () => {
  it("Test 3 - Snapshot vista 'Home' succesful", () => {
    const wrapper = mount(HomeView);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Test 3.1 - Prueba existencia del componente en la ruta", async () => {
    const routes = [
      {
        path: "/",
        name: "home",
        component: () => HomeView,
      },
    ];
    const router = createRouter({
      history: createWebHistory(process.env.BASE_URL),
      routes,
    });

    router.push("/");
    await router.isReady();

    const wrapper = mount(HomeView, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.findComponent(HomeView).exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
