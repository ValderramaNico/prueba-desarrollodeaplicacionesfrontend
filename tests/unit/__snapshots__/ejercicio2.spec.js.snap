// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 2 de vista 'Contact' Test 2.1 - Snapshot vista 'Contact' 1`] = `
<div>
  <h1>contact</h1>
  <nav>
    <router-link to="/">ir al Home</router-link>
  </nav>
</div>
`;

exports[`Test 2 de vista 'Contact' Test2 - Prueba existencia del componente en la ruta 1`] = `
<div>
  <h1>contact</h1>
  <nav><a href="/" class="">ir al Home</a></nav>
</div>
`;

exports[`Test vista 'About' Test 1 - Snapshot vista 'About' succesful 1`] = `
<div>
  <h1>about</h1>
  <nav>
    <router-link to="/">ir al Home</router-link>
  </nav>
</div>
`;

exports[`Test vista 'About' Test 1.2 - Prueba existencia del componente en la ruta 1`] = `
<div>
  <h1>about</h1>
  <nav><a href="/" class="">ir al Home</a></nav>
</div>
`;

exports[`Test vista 'Home' Test 3 - Snapshot vista 'Home' succesful 1`] = `
<div class="home">
  <h1>Pruebas</h1>
  <hr><br>
  <section>
    <h2 class="titles">Ejercicio 1</h2>
    <h2>Contador</h2>
    <p class="counter">0</p>
    <div><button class="btn primary">+1</button><button class="btn warning">-1</button></div>
  </section><br>
  <hr>
  <section>
    <h2 class="titles">Ejercicio 2</h2>
    <h3 for="">Rutas del "Router"</h3>
    <nav>
      <router-link to="/">Home</router-link> | <router-link to="/about">About</router-link> | <router-link to="/contact">Contact</router-link>
    </nav>
  </section><br>
  <hr>
  <section>
    <h2 class="titles">Ejercicio 3</h2>
    <div>
      <div>
        <form action=""><input placeholder="Escriba algo" type="text"><button type="submit">Enviar</button></form>
      </div><br>
      <p>Texto recibido: </p>
    </div>
  </section><br>
  <hr>
  <div class="footer">
    <h2>Prueba - Desarrollo de aplicaciones front-end con framework Vue</h2><br>
    <h3>Realizado por Nicolás Valderrama</h3>
  </div>
</div>
`;

exports[`Test vista 'Home' Test 3.1 - Prueba existencia del componente en la ruta 1`] = `
<div class="home">
  <h1>Pruebas</h1>
  <hr><br>
  <section>
    <h2 class="titles">Ejercicio 1</h2>
    <h2>Contador</h2>
    <p class="counter">0</p>
    <div><button class="btn primary">+1</button><button class="btn warning">-1</button></div>
  </section><br>
  <hr>
  <section>
    <h2 class="titles">Ejercicio 2</h2>
    <h3 for="">Rutas del "Router"</h3>
    <nav><a aria-current="page" href="/" class="router-link-active router-link-exact-active">Home</a> | <a href="/about" class="">About</a> | <a href="/contact" class="">Contact</a></nav>
  </section><br>
  <hr>
  <section>
    <h2 class="titles">Ejercicio 3</h2>
    <div>
      <div>
        <form action=""><input placeholder="Escriba algo" type="text"><button type="submit">Enviar</button></form>
      </div><br>
      <p>Texto recibido: </p>
    </div>
  </section><br>
  <hr>
  <div class="footer">
    <h2>Prueba - Desarrollo de aplicaciones front-end con framework Vue</h2><br>
    <h3>Realizado por Nicolás Valderrama</h3>
  </div>
</div>
`;
